cmake/HalideGeneratorHelpers.cmake

```diff
@@ -148,15 +148,24 @@ function(add_halide_library TARGET)
         message(FATAL_ERROR "Missing FROM argument specifying a Halide generator target")
     endif ()
 
-    if (NOT TARGET ${ARG_FROM})
-        # FROM is usually an unqualified name; if we are crosscompiling, we might need a
-        # fully-qualified name, so add the default package name and retry
-        set(FQ_ARG_FROM "${PROJECT_NAME}::halide_generators::${ARG_FROM}")
-        if (NOT TARGET ${FQ_ARG_FROM})
-            message(FATAL_ERROR "Unable to locate FROM as either ${ARG_FROM} or ${FQ_ARG_FROM}")
-        endif ()
-        set(ARG_FROM "${FQ_ARG_FROM}")
-    endif ()
+    if(ARG_FROM MATCHES ".py$")
+        make_shell_path(PYTHONPATH "$<TARGET_FILE_DIR:Halide::Python>")
+        # TODO: this is ugly, maybe there is a better way?
+        set(GENERATOR_CMD ${CMAKE_COMMAND} -E env PYTHONPATH=${PYTHONPATH} ${Python3_EXECUTABLE} $<SHELL_PATH:${CMAKE_CURRENT_SOURCE_DIR}/${ARG_FROM}>)
+        set(GENERATOR_CMD_DEPS ${ARG_FROM} Halide::Python)
+    else()
+        if (NOT TARGET ${ARG_FROM})
+            # FROM is usually an unqualified name; if we are crosscompiling, we might need a
+            # fully-qualified name, so add the default package name and retry
+            set(FQ_ARG_FROM "${PROJECT_NAME}::halide_generators::${ARG_FROM}")
+            if (NOT TARGET ${FQ_ARG_FROM})
+                message(FATAL_ERROR "Unable to locate FROM as either ${ARG_FROM} or ${FQ_ARG_FROM}")
+            endif ()
+            set(ARG_FROM "${FQ_ARG_FROM}")
+        endif()
+        set(GENERATOR_CMD "$<TARGET_FILE:${ARG_FROM}>")
```

Comment:
`$<TARGET_FILE>` shouldn't be necessary here since the target itself will be the first argument to `add_custom_command`'s `COMMAND` argument. It already handles CMake targets correctly in this case.

Reply:
I'm pretty sure I tried that and it didn't work, but... I'll try again